common, ngram_cache: added const reference for std::pair<> and std::tuple<> more 16 bytes:

GermanAizek opened the pull request:
- std::pair<llama_ngram, llama_ngram_cache_part> (72 bytes -> 8 bytes)
- std::tuple<std::string, float> (40 bytes -> 8 bytes)

GermanAizek committed:
Added const reference for std::pair<> and std::tuple<> more 16 bytes:

- std::pair<llama_ngram, llama_ngram_cache_part> (72 bytes -> 8 bytes)
- std::tuple<std::string, float> (40 bytes -> 8 bytes)

GermanAizek commented:
MacOS test is failed.

JohannesGaessler commented:
Good catch with the `&`. Does the compilation work if you just remove the `const`?

GermanAizek committed:
Merge branch 'ggerganov:master' into const-ref-pair

GermanAizek committed:
Added const reference for std::pair<> and std::tuple<> more 16 bytes:

- std::pair<llama_ngram, llama_ngram_cache_part> (72 bytes -> 8 bytes)
- std::tuple<std::string, float> (40 bytes -> 8 bytes)

GermanAizek commented:
> Good catch with the `&`. Does the compilation work if you just remove the `const`?

@JohannesGaessler, for MacOS again now working only `&` without `const`
More: https://github.com/ggerganov/llama.cpp/actions/runs/9152668091/job/25160468570?pr=7270

GermanAizek committed:
Merge branch 'master' into const-ref-pair

GermanAizek commented:
@mofosyne @JohannesGaessler @Nexesenex @teleprint-me @ggerganov  only CI / macOS-latest-make is failed. Why macos cmake is succesful???
